File: src/command/ClearCommand.ts
import type { Command } from "./Command";
import { Calculator } from "../memento/Calculator";
import { CalculatorMemento } from "../memento/CalculatorMemento";

export class ClearCommand implements Command {
    private calculator: Calculator;
    private previousState?: CalculatorMemento;

    constructor(calculator: Calculator) {
        this.calculator = calculator;
    }

    execute(): void {
        this.previousState = this.calculator.save();
        this.calculator.clear();
    }

    undo(): void {
        if (this.previousState) {
            this.calculator.restore(this.previousState);
        }
    }
}

File: src/main/java/calculadora/command/ClearCommand.java
package calculadora.command;

/**
 *
 * @author vedz11
 */
import calculadora.memento.*;

public class ClearCommand implements Command {
    private final Calculator calculator;
    private CalculatorMemento previousState;

    public ClearCommand(Calculator calculator) {
        this.calculator = calculator;
    }

    @Override
    public void execute() {
        previousState = calculator.save();
        calculator.clear();
    }

    @Override
    public void undo() {
        calculator.restore(previousState);
    }
}